### ISSUE TITLE
Reactive packages (session,etc) without get/set

### ISSUE THREAD

**Issue by janat08** [meteor/meteor-feature-requests#145]
- Array methods return non-results which require proxies for changing state
- Array variant for sessions/vars, or something silly like reactive stuff being preprocessed to arrays as if prepack.io after which you just observe length for new properties, while commiting other object based modifications through array methods.
- One argues that it is hypothetically worth making depends precise enough that it could be used indiscriminately, if using array methods, it's easier to map dependency graph.

**Comment by hwillson** [comment]
I'm quite sure what the feature request here is. Can you provide some pseudo-code examples of what you're thinking?

**Comment by mitar** [comment]
I think they are suggesting that we should not need getter or setter methods, but could use JavaScript proxy to intercept settings and getting values.

See [this discussion in Vue about using proxy](https://github.com/vuejs/vue/issues/3020#issuecomment-223326589).

But we could also use something in between, like Vue is currently using, which is using proper getters and setters on objects, instead of getter and setter methods.

@janat08, maybe you want to check out [reactive field](https://github.com/peerlibrary/meteor-reactive-field) which already provides a simpler way to use reactive vars instead of having to call `get` or `set`, but it is not the same as using getters or setters or proxies.

**Comment by mitar** [comment]
(I am not sure I understand other point they made.)

**Comment by janat08** [comment]
Yes I'm asking that your basic code becomes reactive on get go. I also complain about array methods returning some inner values rather than array which makes sessions ugly to use with them. I then decided to make a case that if you do the "silly" thing, with array for proxy, then depends won't have to be bound to block of code or just the whole variable as reactive var goes. I guess when you pass _id in smart component to reduce reactivity, it would instead make sense to just use proxy, or filter(), as with my suggestion, to insert depends on property that child components exactly cares for if that would work by any measure as depends goes.

@mitar You consistently refer to me as they, but I'm not a legion, "it" will do fine :D.

@hwillson I guess where I'm coming from is that it gets difficult to handle DB derived state that I don't wish to submit back to DB on user input yet it is ultimately derived from there, and all in all v-model is nice as is context that mitar with sashko are willing to kill for it has to be reactivated within template yet again.
